drivers/spi/spi_context.h

```diff
@@ -183,17 +203,44 @@ gpio_dt_flags_t spi_context_cs_active_level(struct spi_context *ctx)
 	return GPIO_ACTIVE_LOW;
 }
 
+static inline int spi_cs_configure(const struct spi_cs_control *cs_gpio)
+{
+	return gpio_pin_configure(cs_gpio->gpio_dev,
+			   cs_gpio->gpio_pin,
+			   cs_gpio->gpio_dt_flags |
+			   GPIO_OUTPUT_INACTIVE);
+}
+
+static int __attribute__((unused)) spi_context_cs_configure_all(struct spi_context *ctx)
+{
+	int err;
+	const struct spi_cs_control *cs_gpio;
+
+	for (cs_gpio = ctx->cs_gpios;
+	     cs_gpio < &ctx->cs_gpios[ctx->num_cs_gpios];
+	     cs_gpio++) {
+		err = spi_cs_configure(cs_gpio);
+		if (err) {
+			return err;
+		}
+	}
+
+	return 0;
+}
+
 static inline void spi_context_cs_configure(struct spi_context *ctx)
 {
+	int ret;
+
 	if (ctx->config->cs && ctx->config->cs->gpio_dev) {
 		/* Validate CS active levels are equivalent */
 		__ASSERT(spi_context_cs_active_level(ctx) ==
 			 (ctx->config->cs->gpio_dt_flags & GPIO_ACTIVE_LOW),
 			 "Devicetree and spi_context CS levels are not equal");
-		gpio_pin_configure(ctx->config->cs->gpio_dev,
-				   ctx->config->cs->gpio_pin,
-				   ctx->config->cs->gpio_dt_flags |
-				   GPIO_OUTPUT_INACTIVE);
+		ret = spi_cs_configure(ctx->config->cs);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure 'cs' gpio: %d", ret);
+		}
```

Comment:
`spi_context_cs_configure` should be modified to return error code, this will hide the problem, e.g. if no logging is enabled.

Reply:
I saw it but I thought that someone had some serious reason to ignore the return code from `gpio_pin_configure` function so I haven't changed that. I will fix it.

Reply:
Fixed.